src/android/NativeRingtones.java

```diff
@@ -22,26 +22,42 @@
 import android.util.Log;
 import android.database.Cursor;
 import android.content.Context;
+import org.apache.cordova.CallbackContext;
+import org.apache.cordova.PluginResult;
+import org.apache.cordova.PluginResult.Status;
 
 /**
  * This class echoes a string called from JavaScript.
  */
 public class NativeRingtones extends CordovaPlugin {
 
     private static MediaPlayer currentRingtone = null;
+    private static HashMap<String, CallbackContext> completeCallbacks;
 
     @Override
     public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
         if (action.equals("get")){
             return this.get(args.getString(0), callbackContext);
         }
-        if (action.equals("play")){
+        else if (action.equals("play")){
             // ringtoneUri, playOnce, volume, streamType
             return this.play(args.getString(0), args.getBoolean(1), args.getInt(2), args.getInt(3), callbackContext);
         }
-        if (action.equals("stop")){
+        else if (action.equals("stop")){
             return this.stop(callbackContext);
         }
+        else if (action.equals("addCompleteListener")) {
+            if (completeCallbacks == null) {
```

Comment:
Copiado de:

https://github.com/we-are-Joinup/cordova-plugin-nativeaudio/blob/master/src/android/NativeAudio.java#L265